File: 0861-score-after-flipping-matrix/0861-score-after-flipping-matrix.py
class Solution:
    def matrixScore(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])
        
        # Set first column
        for i in range(m):
            if grid[i][0] == 0:
                # Flip row
                for j in range(n):
                    grid[i][j] ^= 1 #- grid[i][j] # Bitwise XOR for flip
                    
        # Optimize columns except first column
        for j in range(1, n):
            count_zero = 0
            # Count zeros
            for i in range(m):
                if grid[i][j] == 0:
                    count_zero += 1
            
            # Flip the column if more zeros for better score
            if count_zero > m - count_zero:
                for i in range(m):
                    grid[i][j] ^= 1 # Bitwise XOR for flip
                    
        # Calculate the final score considering bit positions
        ans = 0
        for i in range(m):
            for j in range(n):
                # Left shift bit by place value of column to find column contribution
                column_score = grid[i][j] << (n - j - 1)
                # Add contributio nto score
                ans += column_score
        
        # Return the final result
        return ans
        
        

File: 0861-score-after-flipping-matrix/0861-score-after-flipping-matrix.cpp
class Solution {
public:
    int matrixScore(vector<vector<int>>& grid) {
        int m = grid.size(), n = grid[0].size();
        
        // Set the first column for flip
        for (int i = 0; i < m; ++i){
            if (grid[i][0] == 0){
                for (int j = 0; j < n; ++j){
                    grid[i][j] = 1 - grid[i][j]; // Bitwise XOR for flip
                }
            }
        }
        
        // Optimize the column after the first column
        for (int j = 0; j < n; ++j){
            int countZero = 0;
            for (int i = 0; i < m; ++i) {
                if (grid[i][j] == 0) ++countZero;
            }
            
            if (countZero > m - countZero){
                for (int i = 0; i < m; ++i)
                    grid[i][j] ^= 1; // Bitwise XOR for flip
            }
        }
        
        // Calculate the final score considering bit positions
        int ans = 0;
        for (int i = 0; i < m; ++i){
            int columnScore = 0;
            for (int j = 0; j < n; ++j)
                // Left shift bit by place value of column to find column contribution.
                columnScore += grid[i][j] << (n - j - 1);
            // Add contribution to score
            ans += columnScore;
        }
       
        // return the final ans
        return ans;
    }
};